### PR TITLE
login: smoother mode memory (fixes #4879)

### LINKED ISSUE #4879: darkmode switch in login page
currently only users who are logged in can toggle the app theme in the setting. We need to enable this even without login by having this functionality on the login page

### PR DESCRIPTION
**Pull request by Avinash-Codes**
fixes #4879

### CODE REVIEW

File: app/src/main/java/org/ole/planet/myplanet/MainApplication.kt [hunk 1]
```diff
@@ -101,7 +101,6 @@ class MainApplication : Application(), Application.ActivityLifecycleCallbacks {
                 }
             }
         }
-
```
Comment (Okuro3499) [inline app/src/main/java/org/ole/planet/myplanet/MainApplication.kt:104]: revert this space change

File: app/src/main/java/org/ole/planet/myplanet/ui/SettingActivity.kt [hunk 2]
```diff
@@ -267,36 +270,68 @@ class SettingActivity : AppCompatActivity() {
...
-                ThemeMode.DARK -> 1
-                else -> 2
-            }
+        companion object {
+            fun darkMode(context: Context) {
+                val options = arrayOf(
+                    context.getString(R.string.dark_mode_off),
+                    context.getString(R.string.dark_mode_on),
+                    context.getString(R.string.dark_mode_follow_system)
+                )
```
Comment (Okuro3499) [inline app/src/main/java/org/ole/planet/myplanet/ui/SettingActivity.kt:279]: you could revert this to be one line as before

File: app/src/main/java/org/ole/planet/myplanet/ui/SettingActivity.kt [hunk 3]
```diff
@@ -267,36 +270,68 @@ class SettingActivity : AppCompatActivity() {
...
-                .setSingleChoiceItems(ArrayAdapter(requireContext(), R.layout.checked_list_item, options), checkedItem) { dialog, which ->
-                    val selectedMode = when (which) {
-                        0 -> ThemeMode.LIGHT
-                        1 -> ThemeMode.DARK
-                        2 -> ThemeMode.FOLLOW_SYSTEM
-                        else -> ThemeMode.FOLLOW_SYSTEM
+                val builder = AlertDialog.Builder(context, R.style.CustomAlertDialogStyle)
+                    .setTitle(context.getString(R.string.select_theme_mode))
+                    .setSingleChoiceItems(
+                        ArrayAdapter(
```
Comment (Okuro3499) [inline app/src/main/java/org/ole/planet/myplanet/ui/SettingActivity.kt:290]: this too

**Review by Okuro3499 — CHANGES_REQUESTED** [review]

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -30,6 +30,7 @@
     <string name="menu_logout">Logout</string>
     <string name="menu_feedback">Feedback</string>
     <string name="system_name">myPlanet</string>
+    <string name="select">Select</string>
```
Comment (Okuro3499) [inline app/src/main/res/values/strings.xml:33]: translate this in the other string files too

**Review by Okuro3499 — CHANGES_REQUESTED** [review]

**Review by Okuro3499 — APPROVED** [review]

**Review by deeppp15 — APPROVED** [review]

File: app/src/main/res/values-fr/strings.xml [hunk 5]
```diff
@@ -29,7 +29,8 @@
     <string name="menu_surveys">Sondages</string>
     <string name="menu_logout">Déconnexion</string>
     <string name="menu_feedback">Retour d\'information</string>
-    <string name="system_name">myPlanet</string>
+    <string name="system_name">Sélectionner</string>
+    <string name="select">Select</string>
```
Comment (dogi) [inline app/src/main/res/values-fr/strings.xml:33]: system name is myPlanet

**Review by dogi — CHANGES_REQUESTED** [review]

**Review by dogi — APPROVED** [review]

**Pull request merged**